Update syntax.php

dunron opened the pull request:
A few quick fixes to make the explain plugin work with the newest Dokuwiki ("Igor").

dunron committed:
Update syntax.php

A few quick fixes to make the explain plugin work with the newest Dokuwiki ("Igor").

splitbrain commented on syntax.php:
```diff
@@ -105,7 +105,7 @@ function handle($match, $state, $pos, Doku_Handler $handler) {
         foreach (array_keys($this->map) as $rxmatch) {
             if ($match === $rxmatch ||
-                    ($this->map[$rxmatch]['i'] && utf8_strtolower($match) === $rxmatch)) {
+                    ($this->map[$rxmatch]['i'] && strtolower($match) === $rxmatch)) {
```
This breaks UTF-8 compatibility

dunron replied:
Good point. Would you settle for mb_strtolower then? Not perfect either but it will handle most of UTF-8.

splitbrain replied:
no. please use the appropriate UTF-8 class provided with DokuWiki

dunron committed:
Update syntax.php

Implemented __construct as suggested by jorbig, used Dokuwiki UTF-8 class, improved internal link handling and prevented undefined array warnings in render function.

Tested with 2025-05-14a "Librarian" and PHP 8.3.23

splitbrain commented on syntax.php:
```diff
@@ -76,7 +76,9 @@ function _link($target) {
-        list($id, $hash) = explode('#', $target, 2);
+        $parts = explode('#', $target, 2);
+        $id = $parts[0];
+        $hash = isset($parts[1]) ? $parts[1] : '';
```
can be simplified by using our sexplode() method
